tools/ui-components/src/drop-down/drop-down.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Menu } from '@headlessui/react';
+import { DropDownButton } from './drop-down-button';
+
+type MenuItemsProps = React.ComponentPropsWithoutRef<typeof Menu.Items> & {
+  dropup?: boolean;
+};
+
+export type DropdownProps = React.ComponentPropsWithoutRef<typeof Menu>;
+
+const dropDownItems =
```

Comment:
Can we revert back to an array and use the concatenate method on a separate array for the drop-up?

Something like this:

```js
const dropdownItems = ['value1', 'value2'];
const dropupItems = ['value3'];

const values = [...dropdownItems].concat(dropdownItems);
```

Reply:
The reason that I was using `push` is to mutate the array, so the users don't need to copy the array one time or in your example two times, to see the component.

Yes, we can, but I don't know if we should.

Reply:
I think it was mostly for readability, but sure we can resolve this.

Reply:
True, your suggestion is more readable, and to be honest, I should have put context of why the mutation happens